htmd/qm/custom.py

```diff
@@ -130,8 +130,34 @@ def goalFunc(x):
             grad = -forces.reshape(-1)
             return energy, grad
 
-        res = minimize(goalFunc, coords.reshape(-1), method='L-BFGS-B', jac=True, options={'ftol': 0, 'gtol': 0.05})
-        endcoords = res.x.reshape((natoms, 3)).copy()
+        gtol = 0.1  # kcal/mol/A
+        maxattempts = 50
+        bestres = None
+        bestforces = np.inf
+        for i in range(maxattempts):
+            res = minimize(goalFunc, coords.reshape(-1), method='L-BFGS-B', jac=True, options={'ftol': 0, 'gtol': gtol})
+            maxforcecomp = np.abs(res.jac).max()
+
+            if maxforcecomp < bestforces:
+                bestforces = maxforcecomp
+                bestres = res
+
+            if maxforcecomp > gtol:
```

Comment:
One attempt is one minimization. Why do you need the second one?

Reply:
Resets the hessian

Reply:
The Hessian is reset with each minimization. Still I don't get the point, why do you two per attempt, you could just loop 100 attempts instead.

Reply:
It continues where the previous one quit. See what I pass it as starting
coords. It's the result of the above minimization

Reply:
I had cases where restarting from the previous "near minimum" finished the minimization fine.